client/my-sites/upgrades/domain-management/components/domain/primary-flag.jsx

```diff
@@ -1,27 +1,39 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
 
 /**
  * Internal dependencies
  */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isDomainOnlySite } from 'state/selectors';
+import { localize } from 'i18n-calypso';
 import Notice from 'components/notice';
 
-const DomainPrimaryFlag = React.createClass( {
-	propTypes: {
-		domain: React.PropTypes.object.isRequired
-	},
-
+class DomainPrimaryFlag extends Component {
 	render() {
-		if ( this.props.domain.isPrimary ) {
+		const { isDomainOnly, domain, translate } = this.props;
+
+		if ( domain.isPrimary && ! isDomainOnly ) {
 			return (
-				<Notice isCompact status="is-success">{ this.translate( 'Primary Domain' ) }</Notice>
+				<Notice isCompact status="is-success">{ translate( 'Primary Domain' ) }</Notice>
 			);
 		}
 
 		return null;
 	}
-} );
+}
+
+DomainPrimaryFlag.propTypes = {
+	domain: React.PropTypes.object.isRequired,
+	isDomainOnly: React.PropTypes.bool,
```

Comment:
Thanks for adding `propTypes` to this. I'm just noting here that we need to add `translate` to this list now that it is passed through a prop.